liblepton/src/scheme_object.c

```diff
@@ -719,6 +721,80 @@ SCM_DEFINE (set_object_color_x, "%set-object-color!", 2, 0, 0,
   return obj_s;
 }
 
+
+
+/*! \brief Check whether an object is locked.
+ *
+ * \par Function Description
+ * Check the state of an object's selectable flag: if it's true, the
+ * object is considered to be unlocked, otherwise it is locked.
+ *
+ * \note Scheme API: Implements the %object-locked? procedure in the
+ * (geda core object) module.
+ *
+ * \param obj_s  #OBJECT smob to inspect.
+ *
+ * \return       Boolean value indicating whether \a obj_s is locked.
+ */
+SCM_DEFINE (object_locked_x, "%object-locked?", 1, 0, 0,
```

Comment:
There is a convention about C function naming if their Scheme
counterparts contain special symbols like `!` or `?` in their
names. `!` transforms into `_x` (eXclamation mark), and `?`
transforms into `_p` (`p` just looks a bit like the question
mark). Please fix this in your code.